architecture/core_concepts/builds_and_image_streams.adoc

```diff
@@ -175,29 +175,157 @@ endif::[]
+
+
+For a curated set of image streams, see the
 link:https://github.com/openshift/library[OpenShift Image Streams and Templates
 library].
 
-For example, if a deployment is using a certain image and a new version of that
-image is created, a deployment could be automatically performed.
+When using image streams, it is important to understand what the image stream tag is pointing to and how changes to tags and images can affect you. For example:
+
+* If your image stream tag points to a Docker image tag, you need to understand how that Docker image tag is updated. For example, a Docker image tag `docker.io/ruby:2.4` will probably always point to a v2.4 ruby image. But, a Docker image tag `docker.io/ruby:latest` will probably change with major versions. So, the Docker image tag that a image stream tag points to can tell you how stable the image stream tag will be, if you choose to reference it.
+
+* If your image stream tag follows another image stream tag (it does not point directly to a docker image tag), it is possible that the image stream tag will be updated to follow a different image stream tag in the future. Again, this could result in picking up an incompatible version change.
+
+=== Important terms
+
+Docker repository::
+A collection of related docker images and tags identifying them.
+For example, the OpenShift Jenkins images are in a Docker repository:
++
+----
+docker.io/openshift/jenkins-2-centos7
+----
+
+Docker registry::
+A content server that can store and service images from Docker repositories. 
+For example:
++
+----
+registry.access.redhat.com
+----
+
+Docker image::
+ A specific set of content that can be run as a container. Usually associated with a particular tag within a docker repository.
+
+Docker image tag::
+A label applied to a Docker image in a repository that distinguishes a specific image. For example, here *3.6.0* is a tag:
+
+----
+docker.io/openshift/jenkins-2-centos7:3.6.0
+----
+
+[NOTE]
+====
+A Docker image tag can be updated to point to new Docker image content at any time.
+====
+
+Docker image ID::
+A SHA code that can be used to pull an image. For example:
+
+----
+docker.io/openshift/jenkins-2-centos7@sha256:ab312bda324
+----
+
+[NOTE]
+====
+A SHA image ID cannot change. A specific SHA ID always references the exact same docker image content.
+====
+
+Image stream::
+An {product-title} object that contains pointers to any number of Docker-formatted container images identified by tags. You can think of an image stream
+as equivalent to a Docker repository.
+
+Image stream tag::
+A named pointer to an image in an image stream. An image stream tag is similar to a Docker image tag. See xref:image-stream-tag[Image Stream Tag] below.
+
+Image stream image::
+An image that allows you to retrieve a specific Docker image from a particular image stream where it is tagged.
+See xref:image-stream-image[Image Stream Images] below.
```

Comment:
@bparees Is an _image stream image_ simply an image that is referenced in an image stream?

Reply:
an "imagestreamimage" is an api resource object which pulls together some metadata about a particular image sha.

and yes it mostly exists because there is an imagestream that contains a tag that is currently pointing to that particular image sha.